oauth2/src/main/java/com/okta/spring/boot/oauth/ReactiveOktaOAuth2ResourceServerHttpServerAutoConfig.java

```diff
@@ -55,9 +63,25 @@ BeanPostProcessor oktaOAuth2ResourceServerBeanPostProcessor(OktaOAuth2Properties
         public Object postProcessAfterInitialization(Object bean, String beanName) {
             if (bean instanceof ServerHttpSecurity) {
                 final ServerHttpSecurity http = (ServerHttpSecurity) bean;
-                http.oauth2ResourceServer().jwt()
+
+                try {
+                    if (TokenUtil.isRootOrgIssuer(oktaOAuth2Properties.getIssuer())) {
+                        log.debug("Opaque Token validation/introspection will be configured.");
+                        http.oauth2ResourceServer().opaqueToken();
+                        return http;
+                    }
+                } catch (MalformedURLException ex) {
+                    throw new IllegalArgumentException(ex.getMessage());
+                }
+
+                if (!isEmpty(oktaOAuth2Properties.getClientId()) && !isEmpty(oktaOAuth2Properties.getClientSecret())) {
```

Comment:
This condition could mean we could do JWT as well. I don't think this is a qualifying/sufficient condition to start a Opaque token flow.l

Reply:
@sergiishamrai-okta check this out: https://github.com/okta/okta-spring-boot/pull/213/files